app/controllers/api/v1/users_controller.rb

```diff
@@ -39,28 +39,34 @@ def show
       # POST /api/v1/users.json
       # Creates and saves a new user record in the database with the provided parameters
       def create
-        smtp_enabled = ENV['SMTP_SERVER'].present?
+        return render_error status: :forbidden if external_authn_enabled?
+
         # Check if this is an admin creating a user
         admin_create = current_user && PermissionsChecker.new(current_user:, permission_names: 'ManageUsers', current_provider:).call
 
-        # Users created by a user will have the creator language by default with a fallback to the server configured default_locale.
-        create_user_params[:language] = current_user&.language || I18n.default_locale if create_user_params[:language].blank?
+        # Allow only administrative access for authenticated requests
+        return render_error status: :forbidden if current_user && !admin_create
 
         registration_method = SettingGetter.new(setting_name: 'RegistrationMethod', provider: current_provider).call
 
         if registration_method == SiteSetting::REGISTRATION_METHODS[:invite] && !valid_invite_token && !admin_create
           return render_error errors: Rails.configuration.custom_error_msgs[:invite_token_invalid]
         end
 
-        user = UserCreator.new(user_params: create_user_params.except(:invite_token), provider: current_provider, role: default_role).call
-
-        user.verify! unless smtp_enabled
-
         # TODO: Add proper error logging for non-verified token hcaptcha
         if !admin_create && hcaptcha_enabled? && !verify_hcaptcha(response: params[:token])
           return render_error errors: Rails.configuration.custom_error_msgs[:hcaptcha_invalid]
         end
 
+        # Users created by a user will have the creator language by default with a fallback to the server configured default_locale.
+        create_user_params[:language] = current_user&.language || I18n.default_locale if create_user_params[:language].blank?
+
+        user = UserCreator.new(user_params: create_user_params.except(:invite_token), provider: current_provider, role: default_role).call
+
+        smtp_enabled = ENV['SMTP_SERVER'].present?
+
+        user.verify! unless smtp_enabled
```

Comment:
We can just drop the `smtp_enabled` variable here since it's only used once

Reply:
It can just be `user.verify! unless ENV['SMTP_SERVER'].present?`

Reply:
Okay

Reply:
It's used also in L74,
Do you still wanna drop it?

Reply:
Ah no I missed that. All good then